src/main/java/com/eucalyptus/tests/awssdk/TestRoute53Api.groovy

```diff
@@ -0,0 +1,706 @@
+        }
+      }
+
+      N4j.print( "Tagging hosted zone with ten updated tags ${hostedZoneId}" )
+      changeTagsForResource( new ChangeTagsForResourceRequest(
+          resourceType: 'hostedzone',
+          resourceId: hostedZoneId,
+          addTags: [
+              new Tag(key: 'k0', value: 'v0a'),
+              new Tag(key: 'k1', value: 'v1a'),
+              new Tag(key: 'k2', value: 'v2a'),
+              new Tag(key: 'k3', value: 'v3a'),
+              new Tag(key: 'k4', value: 'v4a'),
+              new Tag(key: 'k5', value: 'v5a'),
+              new Tag(key: 'k6', value: 'v6a'),
+              new Tag(key: 'k7', value: 'v7a'),
+              new Tag(key: 'k8', value: 'v8a'),
+              new Tag(key: 'k9', value: 'v9a'),
+          ]
+      ) )
+
+      N4j.print( "Listing hosted zone to verify tags ${hostedZoneId}" )
+      listTagsForResource( new ListTagsForResourceRequest(
+          resourceType: 'hostedzone',
+          resourceId: hostedZoneId,
+      )).with {
+        N4j.print("Tags: ${resourceTagSet}")
+        Assert.assertNotNull('ResourceTagSet', resourceTagSet)
+        Assert.assertEquals('Resource type', 'hostedzone', resourceTagSet.resourceType)
+        Assert.assertEquals('Resource id', hostedZoneId, resourceTagSet.resourceId)
+        Assert.assertNotNull('ResourceTagSet', resourceTagSet.tags)
+        Assert.assertEquals('resource tag count', 10, resourceTagSet.tags.size())
+        (0..9).each { index ->
+          Assert.assertEquals("resource tag key[${index}]", "k${index}" as String, resourceTagSet.tags.get(index).key)
+          Assert.assertEquals("resource tag value[${index}]", "v${index}a" as String, resourceTagSet.tags.get(index).value)
+        }
+      }
+
+      N4j.print( "Deleting hosted zone single tags ${hostedZoneId}" )
+      changeTagsForResource( new ChangeTagsForResourceRequest(
+          resourceType: 'hostedzone',
+          resourceId: hostedZoneId,
+          removeTagKeys: ['k4']
+      ) )
+
+      N4j.print( "Listing hosted zone to verify tag deleted ${hostedZoneId}" )
+      listTagsForResource( new ListTagsForResourceRequest(
+          resourceType: 'hostedzone',
+          resourceId: hostedZoneId,
+      )).with {
+        Assert.assertNotNull('ResourceTagSet', resourceTagSet)
+        Assert.assertEquals('Resource type', 'hostedzone', resourceTagSet.resourceType)
+        Assert.assertEquals('Resource id', hostedZoneId, resourceTagSet.resourceId)
+        Assert.assertNotNull('ResourceTagSet', resourceTagSet.tags)
+        Assert.assertEquals('resource tag count', 9, resourceTagSet.tags.size())
+      }
+
+      N4j.print( "Deleting hosted zone all tags ${hostedZoneId}" )
+      changeTagsForResource( new ChangeTagsForResourceRequest(
+          resourceType: 'hostedzone',
+          resourceId: hostedZoneId,
+          removeTagKeys: ['k0', 'k1', 'k2', 'k3', 'k4', 'k5', 'k6', 'k7', 'k8', 'k9']
```

Comment:
shouldn't we skip k4 since it was already deleted?

Reply:
This tests idempotent tag delete